pandas/core/reshape.py

```diff
@@ -1161,14 +1161,18 @@ def get_empty_Frame(data, sparse):
             sp_indices = sp_indices[1:]
             dummy_cols = dummy_cols[1:]
         for col, ixs in zip(dummy_cols, sp_indices):
-            sarr = SparseArray(np.ones(len(ixs)),
-                               sparse_index=IntIndex(N, ixs), fill_value=0)
+            # import pdb; pdb.set_trace()
+            sarr = SparseArray(np.ones(len(ixs), dtype=np.int),
+                               sparse_index=IntIndex(N, ixs), fill_value=0,
+                               dtype=np.int)
```

Comment:
these should be the smallest ints that will fit; something like what to_numeric does